core/traffic_class.cc

```diff
@@ -187,21 +187,12 @@ std::vector<TrafficClass *> WeightedFairTrafficClass::Children() const {
 
 bool WeightedFairTrafficClass::AddChild(TrafficClass *child,
                                         resource_share_t share) {
-  if (child->parent_) {
-    return false;
-  }
-
-  int64_t pass = 0;
-  if (!runnable_children_.empty()) {
-    pass = runnable_children_.top().pass_;
-  }
-
-  if (!share) {
+  if (child->parent_ || share == 0) {
     return false;
   }
 
   child->parent_ = this;
-  WeightedFairTrafficClass::ChildData child_data{STRIDE1 / share, pass, child};
+  ChildData child_data{STRIDE1 / share, {NextPass()}, child};
```

Comment:
Since this is initializing a struct with a union, for clarity would you be able to use designated initializers?

Reply:
Yeah, I wanted to do that, but g++5 refused with "non-trivial designated initializers are not supported" 😞

Reply:
Hrm... I was hoping that that wouldn't happen with unions because it'd be smart enough to know that one option among the union is set.